numpy/core/src/npysort/quicksort.c.src

```diff
@@ -69,10 +90,12 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
     @type@ *stack[PYA_QS_STACK];
     @type@ **sptr = stack;
     @type@ *pm, *pi, *pj, *pk;
-    npy_intp depth_limit = npy_get_msb(num) * 2;
+    int depth[PYA_QS_STACK];
+    int * psdepth = depth;
+    int cdepth = NPY_SORT_MIN(PYA_QS_STACK, npy_get_msb(num) * 2);
```

Comment:
the factor 2 is just some random increase (which should be documented) to give the algorithm some slack on not ideal data. The only thing that needs to be done is avoid that the depth is almost linear in number of elements so as long as the limit is constant * log2(n) we should have our worst case.
but the min can now be removed